### PR TITLE
perl: add v5.32.1

### PR DESCRIPTION
**Pull request by alalazo**
Make it such that BigSur is seen as 10.16 rather than 11.1, since the latter causes troubles with Perl configuration

### CODE REVIEW

**Comment by alalazo** [comment]
Built successfully on:

* **Spack:** 0.16.0-1045-d70098eb7b
* **Python:** 3.7.3
* **Platform:** darwin-bigsur-cannonlake
* **Concretizer:** original

File: var/spack/repos/builtin/packages/perl/package.py [hunk 1]
```diff
@@ -39,7 +39,8 @@ class Perl(Package):  # Perl doesn't use Autotools, it should subclass Package
     version('5.31.4', sha256='418a7e6fe6485cc713a86d1227ef112f0bb3f80322e3b715ffe42851d97804a5')
 
     # Maintenance releases (even numbers, recommended)
-    version('5.32.0', sha256='efeb1ce1f10824190ad1cadbcccf6fdb8a5d37007d0100d2d9ae5f2b5900c0b4', preferred=True)
+    version('5.32.1', sha256='efeb1ce1f10824190ad1cadbcccf6fdb8a5d37007d0100d2d9ae5f2b5900c0b4', preferred=True)
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/perl/package.py]: That doesn't look right:
```
$ spack checksum perl 5.32.0 5.32.1
    version('5.32.1', sha256='03b693901cd8ae807231b1787798cf1f2e0b8a56218d07b7da44f784a7caeb2c')
    version('5.32.0', sha256='efeb1ce1f10824190ad1cadbcccf6fdb8a5d37007d0100d2d9ae5f2b5900c0b4')
```

**Review by michaelkuhn — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/perl/package.py (hunk 1 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/perl/package.py]: Thanks @michaelkuhn! Definitely a cut'n paste error when I moved the `preferred` argument.

**Review by haampie — APPROVED** [review]

**Review by michaelkuhn — APPROVED** [review]
Builds successfully on:
* **Spack:** 0.16.0-1046-bf33fe4805
* **Python:** 3.9.1
* **Platform:** linux-fedora33-zen2
* **Concretizer:** original

**Pull request merged**